### DISCUSSION TITLE
Which KiBot version should I use with KiCad 7.0?

### DISCUSSION THREAD

**Discussion by kosma** [INTI-CMNB/KiBot#495] [Q&A]
Reason for asking: some time ago I tried using KiBot on 7.0 and things broke badly. I didn't have time to debug back then; I'd like to try again (because it worked wonders in 5.x) but I'm a bit confused (read: completely lost) as to which version I should use. Should I grab one from Releases? Or from Development? Any advice from someone experienced would be greatly appreciated.

**Comment by set-soft** [comment]
Hi @kosma !

For KiCad 7.0.x use KiBot 1.6.1 or newer (1.6.3 is the current release, 1.6.4 is in development).

As for today I suggest using KiCad 7.0.7. Some KiCad releases contains too much bugs that prevents using various KiBot features (I think 7.0.0, 7.0.3 and 7.0.4 are useless). We don't support the KiCad development snapshots, i.e. 7.99. They break one or more times in a week.

You can always consult the [CHANGELOG](https://github.com/INTI-CMNB/KiBot/blob/dev/CHANGELOG.md). 

Another thing to pay attention is to the [kicad_auto_test](https://github.com/INTI-CMNB/kicad_auto_test) releases: [here](https://github.com/orgs/INTI-CMNB/packages?repo_name=kicad_auto_test)

If you take a look at the [kicad7_auto_full](https://github.com/INTI-CMNB/kicad_auto_test/pkgs/container/kicad7_auto_full/versions) package you'll see that the current **latest** is **1.6.3-1_k7.0.7_d12.1_b3.5.1**. This is KiBot 1.6.3 with KiCad 7.0.7, running on Debian 12.1 with Blender 3.5.1. You'll also see a tag **dev** that is currently pointing to **dev_1.6.4-e88df7f_k7.0.7_d12.1_b3.5.1**.

The **dev** code has some new stuff and fixes, but can occasionally break. But images are generated only when the commit passes all the tests (over than 500 and tested on KiCad 5, 6 and 7). So looking at the above mentioned tag you know that commit **e88df7f** passes the tests. The same information can be seen in the **dev** [commits](https://github.com/INTI-CMNB/KiBot/commits/dev). At this moment you can see that **e88df7f595ce0523270d22c40f6c973668ffbe54** has a green mark, indicating it passed all tests.
